h/search/client.py

```diff
@@ -12,14 +12,15 @@ class Client(object):
     """
     A convenience wrapper around a connection to Elasticsearch.
 
-    Holds a connection object, an index name, and the name of the mapping type.
+    Holds a connection object, an index name, es version, and the name of the mapping type.
 
     :param host: Elasticsearch host URL
     :param index: index name
     :param elasticsearch: Elasticsearch library defaulted to elasticsearch1
     """
 
     def __init__(self, host, index, elasticsearch=elasticsearch1, **kwargs):
+        self._version = '.'.join(str(item) for item in elasticsearch.__version__)
```

Comment:
Was there any particular reason for converting to a string here? I expect that the main use of the `client.version` property will be in comparisons, and for that tuples are better representations because Python will conveniently do item-wise comparisons:

```py
(10,0,0) > (9,) # True
"10.0.0" > "9" # False
```

I notice that [PEP 396](https://www.python.org/dev/peps/pep-0396/#specification) does specify that `__version__` _should_ be a string, but the elasticsearch library can't change that now without breaking any existing checks.

Reply:
If you do make this change, then code switching on the ES version can just do:

```py
if client.version >= (6,):
  # ES 6 code path
else:
  # Older code path
```